### ISSUE TITLE
Backward compatibility issue with existing panels (migration from 1.0.0 to 4.1.0)

### ISSUE THREAD

**Issue by malaman** [GridProtectionAlliance/osisoftpi-grafana#127]
Hi, 

we are using currently gridprotectionalliance-osisoftpi-datasource plugin (1.0.0) with grafana 9.4.7 and want to upgrade the plugin to version 4.1.0.

We have tested the grafana (9.4.7) with the plugin gridprotectionalliance-osisoftpi-datasource (4.1.0) and noticed the following.

Existing panels based on gridprotectionalliance-osisoftpi-datasource data source work for in viewer mode  but have an unexpected error in editor mode (dev console traces are attached)

We are using AF Element attributes as a data source for most of our panels.

It seems that a panel JSON requires `targets.segments` array property in JSON even if "Enable New Data Format" is false in Plugin  Configuration.

Do we need to add targets.segments property to all existing panels or it is possible to use some easier way of migration from 1.0.0 to 4.1.0 ?

![Screenshot 2023-06-08 at 14 22 20](https://github.com/GridProtectionAlliance/osisoftpi-grafana/assets/6554653/3b8d06b1-e587-429d-9f7c-c497236b8da4)
![Screenshot 2023-06-08 at 14 17 29](https://github.com/GridProtectionAlliance/osisoftpi-grafana/assets/6554653/8f459d74-7dcf-4c9b-aae6-fb825beae6d2)
![Screenshot 2023-06-08 at 14 18 08](https://github.com/GridProtectionAlliance/osisoftpi-grafana/assets/6554653/405fb916-7e1e-4f48-a970-3e16b8f6b38c)

Panel JSON (works for the plugin v1.0.0, but does not work for 4.1.0)
```json
{
    "type": "timeseries",
    "title": "Panel Title",
    "datasource": {
      "uid": "59xjClaGk",
      "type": "gridprotectionalliance-osisoftpi-datasource"
    },
    "gridPos": {
      "x": 0,
      "y": 0,
      "w": 12,
      "h": 8
    },
    "id": 8,
    "targets": [
      {
        "datasource": {
          "type": "gridprotectionalliance-osisoftpi-datasource",
          "uid": "59xjClaGk"
        },
        "refId": "A",
        "summary": {
          "types": [
            ""
          ],
          "basis": "EventWeighted",
          "interval": "",
          "nodata": "Null"
        },
        "isPiPoint": false,
        "target": "PIAFServer\\DB_NAME\\ELEMENT1;ELEMENT_Attribute",
        "interpolate": {
          "enable": false
        },
        "recordedValues": {
          "enable": false
        },
        "digitalStates": {
          "enable": false
        },
        "elementPath": "PIAFServer\\DB_NAME\\ELEMENT1",
        "attributes": [
          "ELEMENT_Attribute"
        ],
        "webids": [
          "webidxxx"
        ],
        "webid": "webidxxx"
      }
    ],
    "options": {
      "tooltip": {
        "mode": "single",
        "sort": "none"
      },
      "legend": {
        "showLegend": true,
        "displayMode": "list",
        "placement": "bottom",
        "calcs": []
      }
    },
    "fieldConfig": {
      "defaults": {
        "custom": {
          "drawStyle": "line",
          "lineInterpolation": "linear",
          "barAlignment": 0,
          "lineWidth": 1,
          "fillOpacity": 0,
          "gradientMode": "none",
          "spanNulls": false,
          "showPoints": "auto",
          "pointSize": 5,
          "stacking": {
            "mode": "none",
            "group": "A"
          },
          "axisPlacement": "auto",
          "axisLabel": "",
          "axisColorMode": "text",
          "scaleDistribution": {
            "type": "linear"
          },
          "axisCenteredZero": false,
          "hideFrom": {
            "tooltip": false,
            "viz": false,
            "legend": false
          },
          "thresholdsStyle": {
            "mode": "off"
          }
        },
        "color": {
          "mode": "palette-classic"
        },
        "mappings": [],
        "thresholds": {
          "mode": "absolute",
          "steps": [
            {
              "value": null,
              "color": "green"
            },
            {
              "value": 80,
              "color": "red"
            }
          ]
        }
      },
      "overrides": []
    }
  }
```

**Comment by coderReview** [comment]
@malaman this has been fixed in version 4.2.0. It should be available as soon as Grafana releases the plugin.

**Comment by coderReview** [comment]
Version 4.2.0 has been release.
